.github/workflows/dart_plugin.yml

```diff
@@ -25,10 +25,16 @@ jobs:
         run: |
           dart pub get
 
-      - name: 'Analyze and Format'
+      - name: 'Format, Fix & Analyze'
         run: |
+          dart format .
+          dart fix --apply
           dart analyze --fatal-infos
-          dart format --set-exit-if-changed ./
+
+      # actions/checkout fetches only a single commit in a detached HEAD state. Therefore
+      # we need to pass the current branch, otherwise we can't commit the changes.
+      # GITHUB_HEAD_REF is the name of the head branch. GitHub Actions only sets this for PRs.
+      - run: scripts/commit-formatted-code.sh $GITHUB_HEAD_REF
```

Comment:
`GITHUB_HEAD_REF` 

> The head ref or source branch of the pull request in a workflow run. This property is only set when the event that triggers a workflow run is either pull_request or pull_request_target. For example, feature-branch-1

This will break in the `main` branch and maybe release as well.